RFS/src/main/java/com/rfs/version_es_7_10/ShardMetadataData_ES_7_10.java

```diff
@@ -203,14 +202,14 @@ public FileInfo(
             }
 
             long totalLength = length;
-            long numberOfParts = totalLength / partBytes;
+            long numberOfPartsTemp = totalLength / partBytes;
             if (totalLength % partBytes > 0) {
-                numberOfParts++;
+                numberOfPartsTemp++;
             }
-            if (numberOfParts == 0) {
-                numberOfParts++;
+            if (numberOfPartsTemp == 0) {
+                numberOfPartsTemp++;
             }
-            this.numberOfParts = numberOfParts;
+            this.numberOfParts = numberOfPartsTemp;
```

Comment:
It would be easier to understand if you pulled the calculation lines into a static function that explained why the adjustments are being made.